dbt_semantic_interfaces/implementations/elements/dimension.py

```diff
@@ -17,7 +17,7 @@
 ISO8601_FMT = "YYYY-MM-DD"
 
 
-class DimensionValidityParams(HashableBaseModel):
+class PydanticDimensionValidityParams(HashableBaseModel):
```

Comment:
The `Pyantic<thing>` nomenclature bugs me, because the Pydantic-ness of it seems like an internal implementation detail around how we deal with parsing and serialization rather than something we should broadcast throughout the codebase. Maybe we call these `MetricFlow<thing>` or something to indicate that these are the default implementations that MetricFlow expects? 

On the other hand, if we're going to house different serialization mechanisms in this repo, and that's the key differentiator on the implementations, then this naming approach is excellent.